### PR TITLE
[DF] Unify jitting of Snapshot actions with the rest

### PR DESCRIPTION
**Pull request by eguiraud**
This is a simplification of Snapshot's logic itself and a small performance optimization as we don't invoke the interpreter separately for each Snapshot call but we now take advantage of the same machinery that's already used by all other actions (and that merges all code to be jitted together).

This PR fixes the Snapshot side of [ROOT-9790](https://sft.its.cern.ch/jira/browse/ROOT-9790), "Lazy jitting of Cache and Snapshot".

### CODE REVIEW

**Comment by eguiraud** [comment]
Test failures are unrelated

File: tree/dataframe/inc/ROOT/RDF/RInterface.hxx [hunk 1]
```diff
@@ -2383,44 +2375,29 @@ private:
    /// is the address pointing to the storage of the read/created object in/by
    /// the TTreeReaderValue/TemporaryBranch
    template <typename... ColumnTypes>
-   RResultPtr<RInterface<RLoopManager>> SnapshotImpl(std::string_view treename, std::string_view filename,
+   RResultPtr<RInterface<RLoopManager>> SnapshotImpl(std::string_view fullTreeName, std::string_view filename,
```
Comment (stwunsch) [inline tree/dataframe/inc/ROOT/RDF/RInterface.hxx:2378]: Very minor detail: Doxygen still says `treename`

File: tree/dataframe/inc/ROOT/RDF/RInterface.hxx [hunk 2]
```diff
@@ -2410,13 +2410,15 @@ private:
...
       auto snapHelperArgs = std::make_shared<RDFInternal::SnapshotHelperArgs>(RDFInternal::SnapshotHelperArgs{
          std::string(filename), std::string(dirname), std::string(treename), columnList, options});
-      auto actionPtr =
-         RDFInternal::BuildAction<ColumnTypes...>(validCols, snapHelperArgs, fLoopManager->GetNSlots(), fProxiedPtr,
-                                                  RDFInternal::ActionTags::Snapshot{}, fDefines);
-      fLoopManager->Book(actionPtr.get());
 
-      return RDFInternal::CreateSnapshotRDF(validCols, fullTreename, filename, options.fLazy, *fLoopManager,
-                                            std::move(actionPtr));
+      ::TDirectory::TContext ctxt;
```
Comment (stwunsch) [inline tree/dataframe/inc/ROOT/RDF/RInterface.hxx:2393]: Has the `TContext` some magic behind the ctor or is this variable forgotten?

**Review by stwunsch — APPROVED** [review]
LGTM, perfect!

File: tree/dataframe/inc/ROOT/RDF/RInterface.hxx (hunk 2 above)
Comment (stwunsch) [reply tree/dataframe/inc/ROOT/RDF/RInterface.hxx:2393]: Magic :) https://root.cern/doc/master/classTDirectory_1_1TContext.html

Comment (stwunsch) [reply tree/dataframe/inc/ROOT/RDF/RInterface.hxx:2393]: I think you miss the include for `TDirectory.h`?

Comment (eguiraud) [reply tree/dataframe/inc/ROOT/RDF/RInterface.hxx:2384]: fixed

File: tree/dataframe/inc/ROOT/RDF/RInterface.hxx (hunk 1 above)
Comment (eguiraud) [reply tree/dataframe/inc/ROOT/RDF/RInterface.hxx:2369]: removed the doxygen docs altogether, they did not refer to the current implementation

**Pull request merged**